### ISSUE TITLE
[Extensions] Arrange Components after Merge

### ISSUE THREAD

**Issue by Alero-Awani** [meshery/meshery#12369]
### Current Behavior

When merging designs, the components from the new design overlap and get dropped directly on top of the old components, causing potential layout issues.

### Desired Behavior

It would be better if, during the merge, components could be dropped in a specific location where the user moves their mouse. This would provide more control over the layout, ensuring that merged components are placed exactly where the user intends, preventing unwanted overlaps.

<img width="1680" alt="Screenshot 2024-10-14 at 5 06 31 PM" src="https://github.com/user-attachments/assets/f2f88a62-f5af-47a7-9123-459fbd583944">

### Mockups

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Meshery Build and Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 UI Design For Meshery Extensions [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 📚 [Meshery Extensibility Providers](https://docs.meshery.io/extensibility/providers)
- ⌨️ [Meshery API Docs](https://docs.meshery.io/extensibility/api)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)
-

**Comment by yaten2302** [comment]
@Alero-Awani @leecalcote , may I work on this issue?
Also, could you please guide that how can I replicate this issue, as it would give a better understanding about this.

**Comment by leecalcote** [comment]
@yaten2302, no, thanks. Please seek out issues with the "help wanted" label.
